ui/src/app/pages/data/data-set/genomic-extraction-modal.spec.tsx

```diff
@@ -0,0 +1,117 @@
+import * as React from "react";
+
+import {GenomicExtractionModal} from "./genomic-extraction-modal";
+import {DataSetApiStub} from "testing/stubs/data-set-api-stub";
+import {registerApiClient} from "app/services/swagger-fetch-clients";
+import {DataSetApi, TerraJobStatus} from "generated/fetch";
+import {mount} from "enzyme";
+import {workspaceDataStub} from "testing/stubs/workspaces";
+import {genomicExtractionStore} from "app/utils/stores";
+
+describe('GenomicExtractionModal', () => {
+  let dataset;
+  let datasetApiStub;
+  let testProps;
+  let workspace;
+
+  const component = () => {
+    return <GenomicExtractionModal {...testProps}/>;
+  }
+
+  beforeEach(() => {
+    datasetApiStub = new DataSetApiStub();
+    registerApiClient(DataSetApi, datasetApiStub);
+
+    dataset = datasetApiStub.getDatasetMock;
+    workspace = workspaceDataStub;
+
+    testProps = {
+      dataSet: dataset,
+      workspaceNamespace: workspace.namespace,
+      workspaceFirecloudName: workspace.id,
+      closeFunction: () => {},
+      title: 'Top 10 Egregious Hacks Your Tech Lead Doesn\'t Want You To Know'
+    }
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should render', () => {
+    const wrapper = mount(component());
+    expect(wrapper.exists()).toBeTruthy();
+  });
+
+  it('should show a warning when there is a currently running extract for this dataset', () => {
+    const oneHourAgo = new Date();
```

Comment:
nit: this should be an epoch int since that's the type we store in the store

if you want to do this in one line: `moment().subtract(1, 'hour').unix()`

Reply:
`Date.getTime()` does return an epoch. Although, it's a millisecond epoch and I'm not sure offhand if we store a millisecond epoch or a second epoch. Your one-liner returns a second epoch.